src/include/duckdb_go_extension.h

```diff
@@ -0,0 +1,2109 @@
+//===----------------------------------------------------------------------===//
+//
+//                         DuckDB
+//
+// duckdb_go_extension.h
```

Comment:
lets not check in this Go specific file, but hide the generation of this header behind a flag in the generate_c_api script.